components/layout_2020/flow/mod.rs

```diff
@@ -1409,27 +1411,22 @@ fn justify_self_alignment(containing_block: &ContainingBlock, free_space: Au) ->
 fn solve_inline_margins_for_in_flow_block_level(
     containing_block: &ContainingBlock,
     pbm: &PaddingBorderMargin,
-    inline_size: Length,
+    inline_size: Au,
 ) -> ((Au, Au), Au) {
-    let free_space =
-        containing_block.inline_size - pbm.padding_border_sums.inline - inline_size.into();
-    let mut justification = Au::zero();
+    let free_space = containing_block.inline_size - pbm.padding_border_sums.inline - inline_size;
+    let justification = Au::zero();
     let inline_margins = match (pbm.margin.inline_start, pbm.margin.inline_end) {
         (AuOrAuto::Auto, AuOrAuto::Auto) => {
-            let start = Au::zero().max(free_space / 2);
+            let start = free_space / 2;
             (start, free_space - start)
         },
-        (AuOrAuto::Auto, AuOrAuto::LengthPercentage(end)) => {
-            (Au::zero().max(free_space - end), end)
-        },
+        (AuOrAuto::Auto, AuOrAuto::LengthPercentage(end)) => (free_space - end, end),
```

Comment:
Ditto.

Reply:
I already said this in a previous review, this needs to do safe alignment, don't change it into unsafe alignment.